lib/livebook/intellisense/identifier_matcher.ex

```diff
@@ -806,4 +847,37 @@ defmodule Livebook.Intellisense.IdentifierMatcher do
           documentation: {"text/markdown", info.doc}
         }
   end
+
+  defp beam_lib_chunks(bin, key) do
+    with {:ok, {_, [{^key, value}]}} <- :beam_lib.chunks(bin, [key]) do
+      {:ok, value}
+    end
+  end
+
+  defp cell_id_from_module(node, module) do
+    node
+    |> :erpc.call(fn -> module.module_info(:compile)[:source] end)
+    |> :string.split(~c"#cell:")
+    |> case do
+      [_filename, cell_id] -> {:ok, to_string(cell_id)}
+      _otherwise -> :error
+    end
+  end
+
+  defp keyfind(list, key) do
+    List.keyfind(list, key, 0) || :error
+  end
+
+  defp find_type_value(annotations, name, arity) do
+    annotations
+    |> Enum.filter(&match?({:attribute, _, :type, _}, &1))
+    |> Enum.sort_by(&elem(&1, 1), :asc)
+    |> Enum.find_value(:error, fn
+      {:attribute, anno, :type, {^name, _, vars}} when length(vars) == arity ->
+        {:ok, :erl_anno.line(anno)}
+
+      _ ->
+        nil
+    end)
```

Comment:
This performs fewer traversals and only matching lines, which should be faster.

```suggestion
  defp find_type_value(annotations, name, arity) do
    for {:attribute, anno, :type, {^name, _, vars}} when length(vars) == arity <- annotations do
      :erl_anno.line(anno)
    end
    |> Enum.sort()
    |> case do
      [] -> :error
      [line | _] -> {:ok, line}
    end
```